### DISCUSSION TITLE
State flag confirmation

### DISCUSSION THREAD

**Discussion by hridayahuja44** [async-profiler/async-profiler#978] [General]
I had a query regarding this PR:
[[https://github.com/async-profiler/async-profiler/issues/740](url)](https://github.com/async-profiler/async-profiler/issues/740)

<img width="639" alt="Screenshot 2024-08-28 at 15 40 47" src="https://github.com/user-attachments/assets/a6ca35d4-edaa-4baf-a072-7ad0833d88e7">

If we do not mention the state flag as DEFAULT then does it generate CPU profile as default?

**Comment by apangin** [comment]
If `--state` flag is omitted, all `ExecutionSample` events will be aggregated in a single Flame Graph, regardless of state.

Note that #905 adds a number of CLI improvements. Now, you can shorten `--state RUNNABLE,SLEEPING` to `-s run,sleep` or even use a single `--wall` option. Similarly, `--cpu` is now a shortcut for `--state DEFAULT`.

**Reply by hridayahuja44** [reply]
thank you so much for your help, @apangin
